cairo/crates/pragma_feed_types/src/feed.cairo

```diff
@@ -0,0 +1,56 @@
+use pragma_feed_types::{AssetClass, AssetClassId, FeedType, FeedTypeId};
+use pragma_maths::felt252::{FeltBitAnd, FeltDiv};
+
+pub type PairId = felt252;
+pub type FeedId = felt252;
+
+#[derive(Debug, Clone, Drop, PartialEq, Serde)]
+pub struct Feed {
+    pub asset_class: AssetClass,
+    pub feed_type: FeedType,
+    pub pair_id: PairId,
+}
+
+pub trait FeedTrait {
+    /// Try to construct a Feed from the provided FeedId.
+    fn from_id(id: FeedId) -> Option<Feed>;
+
+    /// Returns the id of the Feed.
+    fn id(self: @Feed) -> FeedId;
+}
+
+pub impl FeedTraitImpl of FeedTrait {
+    /// Try to construct a Feed from the provided FeedId.
+    fn from_id(id: FeedId) -> Option<Feed> {
+        let asset_class_felt = id / 0x100000000000000000000000000000000;
+        let asset_class: AssetClass = asset_class_felt.try_into().unwrap();
+
+        // Extract feed_type (middle 64 bits)
+        let feed_type_felt = (id / 0x100000000000000) & 0xFFFFFFFFFFFFFFFF;
+        let feed_type: FeedType = feed_type_felt.try_into().unwrap();
+
+        // Extract pair_id (remaining bits, maximum 28 bytes)
+        let pair_id = id
```

Comment:
we could add a check to make sure that the pair id does not exceed 28 bytes here, and revert if it overflows. Also if I recall correctly, if the `try_into`conversion fails, it will just display Option:: Unwrap failed as an error. It could be interesting to implement proper error management here instead.

Reply:
Just realized `Result` also exists in Cairo... used them, we now have proper errors 🫡